cmd/producer/main.go

```diff
@@ -82,57 +80,46 @@ if async is preferred, then write request to redis.
 if symnchronous is preferred, then proxy the request.
 */
 func checkHeaderAndServe(w http.ResponseWriter, r *http.Request) {
-	target := &url.URL{
-		Scheme:   "http",
-		Host:     r.Host,
-		RawQuery: r.URL.RawQuery,
+	// if request body exists, check that length doesn't exceed limit
+	requestSizeInt, err := strconv.Atoi(env.RequestSizeLimit)
+	if err != nil {
+		log.Fatal("Error parsing request size string to integer")
 	}
-	if r.Header.Get("Prefer") == "respond-async" {
-		// if request body exists, check that length doesn't exceed limit
-		requestSizeInt, err := strconv.Atoi(env.RequestSizeLimit)
-		if err != nil {
-			log.Fatal("Error parsing request size string to integer")
-		}
-		if r.Body != nil {
-			r.Body = http.MaxBytesReader(w, r.Body, int64(requestSizeInt))
-		}
-		// write the request into buff
-		var buff = &bytes.Buffer{}
-		if err := r.Write(buff); err != nil {
-			if err.Error() == "http: request body too large" {
-				w.WriteHeader(500)
-			} else {
-				log.Print("Error writing to buffer: ", err)
-				w.WriteHeader(500)
-			}
-			return
-		}
-		// translate to string, then json including an id.
-		reqString := buff.String()
-		id := gouuidv6.NewFromTime(time.Now()).String()
-		reqData := requestData{
-			ID:      id,
-			Request: reqString,
-		}
-		reqJSON, err := json.Marshal(reqData)
-		if err != nil {
+	if r.Body != nil {
+		r.Body = http.MaxBytesReader(w, r.Body, int64(requestSizeInt))
```

Comment:
Just to check, can r.Body ever be nil?

Reply:
I think so -- according to go docs:
```
	// Body is the request's body.
	//
	// For client requests, a nil body means the request has no
	// body, such as a GET request. The HTTP Client's Transport
	// is responsible for calling the Close method.
	//
	// For server requests, the Request Body is always non-nil
	// but will return EOF immediately when no body is present.
	// The Server will close the request body. The ServeHTTP
	// Handler does not need to.
```

Reply:
but isn't this is a server request so the third para applies?

Reply:
Agree that this is a server request. updated to remove nil check and updated test as well. Thanks @julz!